opensrp-path/src/main/java/org/ei/opensrp/path/activity/HIA2ReportsActivity.java

```diff
@@ -286,18 +200,34 @@ public void startMonthlyReportForm(String formName, Date date) {
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         if (requestCode == REQUEST_CODE_GET_JSON) {
             if (resultCode == RESULT_OK) {
-
                 try {
                     showFragment = true;
                     String jsonString = data.getStringExtra("json");
+                    JSONObject form = new JSONObject(jsonString);
+                    String monthString = form.getString("report_month");
+                    Date month = HIA2Service.dfyymmdd.parse(monthString);
 
                     JSONObject monthlyDraftForm = new JSONObject(jsonString);
                     Map<String, String> result = JsonFormUtils.sectionFields(monthlyDraftForm);
-                    VaccinatorApplication.getInstance().monthlyTalliesRepository().save(result, report_month);
-                } catch (Exception e) {
+                    boolean saveClicked = false;
+                    if (result.containsKey(FORM_KEY_CONFIRM)) {
+                        try {
+                            saveClicked = Boolean.valueOf(result.get(FORM_KEY_CONFIRM));
+                        } catch (Exception e) {
```

Comment:
I was going with handling all exceptions regardless the same way (in anticipation). To prevent crashes in the background bubbling up to the UI causing the app to stop.